scripts/gen-docs.py

```diff
@@ -0,0 +1,382 @@
+
+    # Parse the file
+    translation_unit = index.parse(file_path)
+
+    # Check for parsing errors
+    if not translation_unit:
+        print(f"Error parsing {file_path}")
+        return []
+
+    functions = []
+
+    # Helper function to recursively traverse the AST
+    def traverse_ast(cursor, parent=None):
+        # Check if the cursor represents a function declaration
+        if cursor.kind == CursorKind.FUNCTION_DECL:
+            # Get function return type
+            return_type = cursor.type.get_result().spelling
+
+            # Get function name
+            func_name = cursor.spelling
+
+            # Get function parameters
+            params = []
+            for param in cursor.get_arguments():
+                params.append({"name": param.spelling, "type": param.type.spelling})
+
+            # Get function location
+            location = cursor.location
+            file_path = location.file.name if location.file else "Unknown"
+            line = location.line
+            column = location.column
+
+            # Check if the function has a body
+            has_body = any(
+                c.kind == CursorKind.COMPOUND_STMT for c in cursor.get_children()
+            )
+
+            # Determine if it's a declaration or definition
+            func_type = "definition" if has_body else "declaration"
+
+            # Add function info to our list
+            functions.append(
+                {
+                    "name": func_name,
+                    "return_type": return_type,
+                    "parameters": params,
+                    "location": {"file": file_path, "line": line, "column": column},
+                    "type": func_type,
+                }
+            )
+
+        # Recursively process children
+        for child in cursor.get_children():
+            traverse_ast(child, cursor)
+
+    # Start traversing from the translation unit cursor
+    traverse_ast(translation_unit.cursor)
+
+    return functions
+
+
+def extract_kernel_docs(
+    project_path: str, kernel_config: Dict[str, Any]
+) -> List[Dict[str, Any]]:
+    """Extract documentation from kernel source files."""
+    kernel_info = []
+    source_files = []
+
+    # Collect source files from kernel config first
+    for kernel_name, kernel_data in kernel_config.items():
+        if "src" in kernel_data and isinstance(kernel_data["src"], list):
+            for src_file in kernel_data["src"]:
+                file_path = os.path.join(project_path, src_file)
+                if os.path.exists(file_path):
+                    source_files.append(Path(file_path))
+
+    # Also collect all other source files
+    for ext in [".cu", ".h", ".cpp", ".cuh"]:
+        extra_files = []
+        for p in Path(project_path).glob(f"**/*{ext}"):
+            # avoid adding `torch-ext` files
+            if "torch-ext" in str(p):
+                continue
+            if p not in source_files:
+                extra_files.append(p)
```

Comment:
Is there any reason for doing this? As far as `kernel-builder` is concerned, unspecified source files do not exist (are not added to the sandbox).